### PR TITLE
Extended crop-thumbnails to fire imagify WebP optimisation on crop.

### PR DESCRIPTION
**Pull request by aaronsummers**
I have included a function to fire the image optimisation provided by Imagify when a thumbnail has been cropped. The community has widely requested this feature.

### CODE REVIEW

File: functions/extend/ImagifyExtension.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * @link https://github.com/aaronsummers
+ * @version 1.0.0
+ * @since 1.8.0
+ * @package crop-thumbnails
+ */
+class ImagifyExtension
+{
+	 public static function doExtend() {
+
+		// crop_thumbnails_after_save_new_thumb is a hook that is triggered after a new thumbnail is created amd returns the file path of the new thumbnail (not the URL)
```
Comment (vollyimnetz) [inline functions/extend/ImagifyExtension.php:19]: @aaronsummers thank you for your code.

Can you please make some small adjustments?
The add_action is always triggered no matter what the Imagify_Auto_Optimization exists.

I would love to see the something linke this:
```
add_action('plugins_loaded', function() {
  if( class_exists( Imagify_Auto_Optimization::class) ) {
    add_action( 'crop_thumbnails_after_save_new_thumb', array( self::class, 'actionDoWebPConvertion' ), 100, 1 );
  }
});
```

I know you checked in actionDoWebPConvertion, but this way not even the action-hook is triggered if the plugin do not exists.

**Review by vollyimnetz — CHANGES_REQUESTED** [review]

File: functions/extend/ImagifyExtension.php (hunk 1 above)
Comment (aaronsummers) [reply functions/extend/ImagifyExtension.php]: This has been updated.

Comment (aaronsummers) [reply functions/extend/ImagifyExtension.php]: Not sure if you need me to mark this as resolved @vollyimnetz ?

Comment (vollyimnetz) [reply functions/extend/ImagifyExtension.php]: I am on it.

**Review by vollyimnetz — APPROVED** [review]

**Pull request merged**